### ISSUE TITLE
Be able to record automation

### ISSUE THREAD

**Issue by bl0b** [bl0b/renoise-xtouch-xctl.xrnx#2]
There is no way currently to record automation from the X-Touch. It would be nice to be able to.
#3 
#4

- [X] Automation page
- [X] Implement record logic and all write modes
- [x] Allow editing the same lane across patterns seamlessly
- [x] Keep lanes set to recording visible on the X-Touch when scrolling left/right
- [x] Document mappings

**Comment by bl0b** [comment]
a8ddb2a838f3839ba34c6c562d829205292c906d

**Comment by nickcent** [comment]
So, does this mean that we are only able to set parameters to own liking but not write automation? What if one holds down the write button while moving a fader? Wouldn't that be the same as right click and hold the mouse?

**Comment by bl0b** [comment]
No, you can record automation already. Switch to the automation page with `ENCODER ASSIGN / TRACK`. I haven't closed the issue yet because of the "continuity of edits" issues when switching patterns or just moving the frame left and right. All 4 write modes are implemented. Only `GROUP` is bogus still.

**Comment by nickcent** [comment]
I'm trying to figure out how to write automation. I have set which parameters I would like to automate, they turn RED but the faders don't do anything. Also, I hit WRITE, then Play, switch to PLUG-IN mode and move the parameters, but they don't get written... ?

**Comment by bl0b** [comment]
@nickcent 
Yeah, there's 3 stages to record automation, sorry (not sorry) :)

0. Go to automation page
1. Select the write mode you want to use (`READ/OFF` turns editing off completely).
2. Press `REC` on the lanes you want to record. Ultimately the recording lanes will remain visible on the X-Touch when you change pattern or scroll the frame. This is also to ensure that you don't edit a neighbouring lane by accident.
3. Touch the fader to edit.

The order of the steps doesn't matter much.

For instance, to overwrite/erase a range, select `WRITE`, press `REC` on the proper X-Touch channel, and touch the fader (without moving it) to erase. As soon as you release the fader, it will stop overwriting.
So, in summary:
 - write mode defines how the faders will modify the automation data, See #4 for details.
 - `REC` enables editing a lane,
- Touching the fader starts an edit, and releasing the fader stops the edit.

**Comment by bl0b** [comment]
descriptions of bindings added in 46e2c291fd8930cc32a8c4351b94bfffeaae59db.
